### PR TITLE
feat(Baseball-Reference): add Presence

### PR DESCRIPTION
**Pull request by EdwinH7**
## Description 

Add presence for Baseball-Reference
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

Screenshot_20221206_045355

Screenshot_20221206_045412

Screenshot_20221206_045426

Screenshot_20221206_045521

</details>

### CODE REVIEW

File: websites/B/Baseball-Reference/metadata.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"id": "143542273609564161",
+		"name": "DjDrake8#2047"
```
Comment (DiscordAnaxes) [inline websites/B/Baseball-Reference/metadata.json:5]: Using old PMD, remove the discriminator (#2047)

File: websites/B/Baseball-Reference/presence.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		presenceData.state = `${
+			document.querySelector("div#meta div h1 span").textContent
+		} | ${document
+			.querySelector("div#meta div p strong")
+			.nextSibling.textContent.replace(/\s/g, " ")
+			.trim()}`;
+		if (!presence.getActivity().largeImageKey)
+			presenceData.largeImageKey = image;
+
+		presenceData.buttons = [{ label: "View Player", url: document.URL }];
```
Comment (theusaf) [inline websites/B/Baseball-Reference/presence.ts:28]: Prefer `href` over `document.URL`

File: websites/B/Baseball-Reference/presence.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+		if (
+			document.querySelector(
+				"#meta > div:nth-child(2) > p:nth-child(5) > a"
+			) !== null
+		) {
+			team = document.querySelector(
+				"#meta > div:nth-child(2) > p:nth-child(5) > a"
+			).textContent;
+			presenceData.state = `${presenceData.state}\nFor The ${team}`;
```
Comment (theusaf) [inline websites/B/Baseball-Reference/presence.ts:38]: I'm not sure if newline characters work in presence states.

Comment (DarkVillager) [reply websites/B/Baseball-Reference/presence.ts:38]: > I'm not sure if newline characters work in presence states.

It doesn't work, just tested it.

File: websites/B/Baseball-Reference/presence.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
+const presence = new Presence({
+		clientId: "1049610101725605899",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	loc = document.location;
```
Comment (theusaf) [inline websites/B/Baseball-Reference/presence.ts:5]: Destructure like so to improve simplicity.
```suggestion
	{ href, pathname, hostname } = document.location;
```

File: websites/B/Baseball-Reference/presence.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+		} ${
+			document.querySelector("div#meta div h1 span").nextElementSibling
+				.textContent
+		}`;
+		if (!presence.getActivity().largeImageKey)
+			presenceData.largeImageKey = image;
+
+		presenceData.buttons = [{ label: "View Team", url: loc.href }];
+	} else if (loc.pathname.includes("/leagues")) {
+		presenceData.details = "Viewing Stats:";
```
Comment (theusaf) [inline websites/B/Baseball-Reference/presence.ts:60]:
```suggestion
		presenceData.details = "Viewing Stats";
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/B/Baseball-Reference/presence.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+		presenceData.largeImageKey = image;
+		presenceData.buttons = [{ label: "View Box Score", url: href }];
+	} else if (pathname.includes("/friv")) {
+		presenceData.details = "Viewing Misc";
+		presenceData.state = document.querySelector("div h1").textContent;
+		presenceData.buttons = [{ label: "View Page", url: href }];
+	} else if (pathname.includes("/oracle")) {
+		presenceData.details = "Consulting the Oracle of Baseball";
+		presenceData.state = document.querySelector("div h1").textContent;
+		presenceData.buttons = [{ label: "View Page", url: href }];
```
Comment (Bas950) [inline websites/B/Baseball-Reference/presence.ts:88]: It seems like there may be a chance no largeImage is shown, you might want to change it so it always has a default

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/B/Baseball-Reference/presence.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
+const presence = new Presence({
+		clientId: "1049610101725605899",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	{ href, pathname, hostname } = document.location;
+
+let team: string = null,
```
Comment (EncryptedDev) [inline websites/B/Baseball-Reference/presence.ts:7]: Why can't `team` be a const?

File: websites/B/Baseball-Reference/presence.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+	image = document.images.item(1).src;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		startTimestamp: browsingTimestamp,
+	};
+	presence.setActivity(presenceData);
```
Comment (EncryptedDev) [inline websites/B/Baseball-Reference/presence.ts:14]: This is at the end of the event as well so not needed here

File: websites/B/Baseball-Reference/presence.ts [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+		if (pathname === "/") presenceData.details = "Viewing Home Page";
+	} else if (pathname.includes("/players")) {
+		presenceData.details = "Viewing a Player";
+		presenceData.state = `${
+			document.querySelector("div#meta div h1 span").textContent
+		} | ${document
+			.querySelector("div#meta div p strong")
+			.nextSibling.textContent.replace(/\s/g, " ")
+			.trim()}`;
+		if (!presence.getActivity().largeImageKey)
```
Comment (EncryptedDev) [inline websites/B/Baseball-Reference/presence.ts:25]: This check isn't needed tbh and`getActivity()` will be deprecated soon

File: websites/B/Baseball-Reference/presence.ts [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+			).textContent;
+			presenceData.state = `${presenceData.state} For The ${team}`;
+		}
+	} else if (pathname.includes("/managers")) {
+		presenceData.details = "Viewing a Manager";
+		presenceData.state = document.querySelector(
+			"div#meta div h1 span"
+		).textContent;
+		if (!presence.getActivity().largeImageKey)
+			presenceData.largeImageKey = image;
```
Comment (EncryptedDev) [inline websites/B/Baseball-Reference/presence.ts:44]: And here

File: websites/B/Baseball-Reference/presence.ts [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+		presenceData.buttons = [{ label: "View Player", url: href }];
+
+		if (
+			document.querySelector("#meta > div:nth-child(2) > p:nth-child(5) > a")
+		) {
+			team = document.querySelector(
+				"#meta > div:nth-child(2) > p:nth-child(5) > a"
+			).textContent;
+			presenceData.state = `${presenceData.state} For The ${team}`;
```
Comment (EncryptedDev) [inline websites/B/Baseball-Reference/presence.ts:36]: I think an unexpected output will be caused by doing this, can you prove to me it won't through a screenshot?

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**